FirebaseSegmentation/Sources/SEGContentManager.m

```diff
@@ -0,0 +1,145 @@
+// Copyright 2019 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "SEGContentManager.h"
+
+#import <FirebaseCore/FIRAppInternal.h>
+#import <FirebaseInstallations/FirebaseInstallations.h>
+#import "FIRSegmentation.h"
+#import "SEGDatabaseManager.h"
+#import "SEGNetworkManager.h"
+#import "SEGSegmentationConstants.h"
+
+NSString *const kErrorDescription = @"ErrorDescription";
+
+@interface SEGContentManager () {
+  NSMutableDictionary<NSString *, id> *_associationData;
+  NSString *_instanceIdentifier;
+  NSString *_instanceIdentifierToken;
+  SEGDatabaseManager *_databaseManager;
+  SEGNetworkManager *_networkManager;
+}
+@end
+
+@implementation SEGContentManager
+
++ (instancetype)sharedInstanceWithOptions:(FIROptions *)options {
+  static dispatch_once_t onceToken;
+  static SEGContentManager *sharedInstance;
+  dispatch_once(&onceToken, ^{
+    sharedInstance = [[SEGContentManager alloc]
+        initWithDatabaseManager:[SEGDatabaseManager sharedInstance]
+                 networkManager:[[SEGNetworkManager alloc] initWithFIROptions:options]];
+  });
+  return sharedInstance;
+}
+
+- (instancetype)initWithDatabaseManager:databaseManager networkManager:networkManager {
+  self = [super init];
+  if (self) {
+    // Initialize the database manager.
+    _databaseManager = databaseManager;
+
+    // Initialize the network manager.
+    _networkManager = networkManager;
+
+    // Load all data from the database.
+    [_databaseManager createOrOpenDatabaseWithCompletion:^(BOOL success, NSDictionary *result) {
+      self->_associationData = [result mutableCopy];
+    }];
+    // TODO(dmandar) subscribe to FIS notifications once integrated.
+  }
+  return self;
+}
+
+// TODO(dmandar) IID only supports default instance. Modify for FIS.
+- (FIRInstallations *)installationsForApp:(NSString *)firebaseApp {
+  return [FIRInstallations installations];
```

Comment:
Note: Make sure this is addressed in the CL that adds Installations.

Reply:
addressed